### PR TITLE
feat: implement estimateBackground() 1st version

### LINKED ISSUE #52: PeakIntegrator estimateBackground()
https://github.com/OpenMS/OpenMS/blob/develop/src/openms/include/OpenMS/ANALYSIS/OPENSWATH/MRMTransitionGroupPicker.h#L529

### PR DESCRIPTION
**Pull request by pcolaianni**
closes #52 

Not a final version, but it works. `baseline_type_` feature is missing. Need to talk about it with you to clarify its behaviour.

I will implement `MSSpectrum` support in all methods at the end, when the three methods are all implemented for `MSChromatogram`s.

### CODE REVIEW

File: src/openms/source/ANALYSIS/OPENSWATH/PeakIntegrator.cpp [hunk 1]
```diff
@@ -45,6 +45,26 @@ namespace OpenMS
 
   PeakIntegrator::~PeakIntegrator() {}
 
+  double PeakIntegrator::estimateBackground(
```
Comment (dmccloskey) [inline src/openms/source/ANALYSIS/OPENSWATH/PeakIntegrator.cpp:48]: The `integration_type_` parameter will determine how the baseline is calculated.  I could see an if/else loop where based on the value, the background type is estimated.

File: src/openms/source/ANALYSIS/OPENSWATH/PeakIntegrator.cpp [hunk 2]
```diff
@@ -45,6 +45,26 @@ namespace OpenMS
...
+    const double intensity_l = chromatogram.RTBegin(left)->getIntensity();
+    const double intensity_r = (chromatogram.RTEnd(right)-1)->getIntensity();
+    const double delta_int = intensity_r - intensity_l; // sign will determine line direction
+    const double delta_input_rt = right - left;
+    double background = 0.0;
+    for (auto it=chromatogram.RTBegin(left); it!=chromatogram.RTEnd(right); ++it)
+    {
+      // calculate the background using the formula
+      // y = mx + b where x = retention time, m = slope, b = left intensity
+      background += delta_int / delta_input_rt * (it->getRT() - left) + intensity_l;
```
Comment (dmccloskey) [inline src/openms/source/ANALYSIS/OPENSWATH/PeakIntegrator.cpp]: Be sure to add in the other cases as well based on the integration_type.  I believe in the code the uncommented version was for intensity_sum while the commented version was for trapezoid

**Review by dmccloskey — COMMENTED** [review]
I made some comments on the behavior of estimateBackground.  It depends not just on the background type but also the integration type

**Comment by pcolaianni** [comment]
I still have to implement the `baseline_type_` feature.

At the moment the method does the following:
if integration type is trepezoid or simpson
then apply trapezoid background estimation
otherwise apply the method that was already implemented in the previous commit (intensity sum)

File: src/openms/source/ANALYSIS/OPENSWATH/PeakIntegrator.cpp [hunk 3]
```diff
@@ -53,18 +53,99 @@ namespace OpenMS
...
+        if (peak_height < it->getIntensity())
+        {
+          peak_height = it->getIntensity();
+          peak_apex_pos = it->getRT();
+        }
+      }
+    }
+    else if (getIntegrationType() == "simpson")
+    {
+      if (n_points < 3 || !(n_points % 2))
```
Comment (dmccloskey) [inline src/openms/source/ANALYSIS/OPENSWATH/PeakIntegrator.cpp]: I can see this part of the if clause causing a lot of headache: `!(n_points % 2)`.  Would it be possible for the case that n_points is even to calculate the peak area with 1) 1 additional point on the right and 2) with 1 less point on the right and then 3) take the average to approximate what the area is?

**Review by dmccloskey — COMMENTED** [review]
I made a note about the simpson's rule when there is an even number of points.  I had not considered this earlier, but we would need to accomodate that case somehow.  I made a potential suggestion.  Please let me know how it sounds to you.

File: src/openms/source/ANALYSIS/OPENSWATH/PeakIntegrator.cpp (hunk 3 above)
Comment (pcolaianni) [reply src/openms/source/ANALYSIS/OPENSWATH/PeakIntegrator.cpp]: Sure, we can do this, assuming there is a `Peak` after `right` position (I guess it's safe to assume so).
I'll make this in another branch/pr

**Pull request merged**